src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -222,6 +225,106 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         Automation::AutomationProperties::SetFullDescription(UseDesktopImageCheckBox(), unbox_value<hstring>(backgroundImgCheckboxTooltip));
 
         Automation::AutomationProperties::SetName(DeleteButton(), RS_(L"Profile_DeleteButton/Text"));
+
+        if (!_FontList)
+        {
+            _UpdateFontList();
+        }
+    }
+
+    void Profiles::_UpdateFontList()
+    try
+    {
+        // initialize font list
+        _FontList = single_threaded_observable_vector<hstring>();
+
+        // get a DWriteFactory
+        ::IUnknown* unknownFactory;
+        THROW_IF_FAILED(DWriteCreateFactory(
+            DWRITE_FACTORY_TYPE_SHARED,
+            __uuidof(::IDWriteFactory),
+            &unknownFactory));
```

Comment:
This is an absolute shoo-in for `winrt::capture`, so I would block for that reason alone :)

Reply:
Unfortunately, `capture` requires the last parameter to be `void**`. Since `DWriteCreateFactory()` requires an `IUnknown`, it's more restrictive and we can't use `capture`. Going with @zadjii-msft's  suggestion instead.